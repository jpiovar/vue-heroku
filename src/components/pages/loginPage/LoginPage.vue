<template>
<div class="container">
    <div class="row align-items-center justify-content-center">
        <div class="col-md-6">
          <LoginForm/>
        </div>
        <a href='#' @click="login">login</a>
    </div>
</div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import { namespace } from 'vuex-class';
import axios, { AxiosResponse } from 'axios';
import { USER } from '../../../stores/constants';
import LoginForm from './components/LoginForm.vue';
import { userProfiles } from '../../../assets/data';

const UserStore = namespace(USER);

@Component({
  components: {
    LoginForm,
  },
  props: {
    msg: {
      required: false,
      type: String,
    },
  },
})
export default class LoginPage extends Vue {
  msg!: string;

  @UserStore.Action loginUser: any;

  login() {
    console.log('login clicked');
    const up = userProfiles.filter(item => item.name === 'janko' && item.password === 'janko');
    if (up) {
      this.loginUser(up[0]);
    }
    // axios
    //   .get('/src/assets/userData.json')
    //   .then(response => this.loginUser());
  }
}

</script>

<style lang="scss" scoped>
.row {
    height: 500px;
    .col-md-6 {
        height: 500px;
    }
}
</style>
