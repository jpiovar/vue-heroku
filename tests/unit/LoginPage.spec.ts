import { shallowMount, mount, Wrapper, createLocalVue } from '@vue/test-utils';
import LoginPage from '@/components/pages/loginPage/LoginPage.vue';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);

const actions = {
    loginUser: jest.fn(),
    storeQuoraData: jest.fn()
};

const store = new Vuex.Store({
    actions
});

const props = {
    author: 'Juraj Piovar',
};

const getWrapperShallow = (propsData = props): Wrapper<LoginPage> => shallowMount(LoginPage, {
    propsData: {
        ...propsData
    },
});

const getWrapperMount = (propsData = props): Wrapper<LoginPage> => mount(LoginPage, {
    propsData: {
        ...propsData
    },
    stubs: ['LoginForm']
});

describe('LoginPage.vue', () => {
    it('should match the snapshot', () => {
        const wrapper: Wrapper<LoginPage> = getWrapperShallow();
        expect(wrapper.vm.$el).toMatchSnapshot()
    });
    it('renders footer exists and visible', () => {
        const wrapper: Wrapper<LoginPage> = getWrapperShallow();
        const spanFooter = wrapper.find('span#footer');
        expect(spanFooter.exists()).toBe(true);
        expect(spanFooter.isVisible()).toBe(true);
    });
    it('renders footer with correct text', () => {
        const wrapper: Wrapper<LoginPage> = getWrapperShallow();
        const spanFooter = wrapper.find('span#footer');
        const year = '2019';
        expect(spanFooter.text()).toBe(`${props.author} @ ${year}`);
    });
    it('renders component id and footer with correct text', () => {
        const wrapper: Wrapper<LoginPage> = getWrapperShallow();
        const mockData = { 'idComponent': 'loginPageIdComponent', 'year': '07/2019' };
        wrapper.setData(mockData); // sets component's data/instance attributes id, year with mock values
        const spanFooter = wrapper.find('span#footer');
        expect(wrapper.attributes().id).toBe('loginPageIdComponent'); // checks if component's DOM id is equal to value of set component's data/instance attribute id
        expect(spanFooter.text()).toBe(`${props.author} @ 07/2019`);
    });
    // it('calls store action "loginUser" when login submit button "loginForm" is clicked and "loginHandle" called', () => {
    //     const wrapper: any = mount(LoginPage, { propsData: { ...props}, store, localVue });
    //     // spyOn(wrapper.vm, 'loginHandle');
    //     // wrapper.find("form").trigger("submit.prevent");
        

    //     wrapper.vm.loginHandle({name: 'janko', password: 'janko'});
    // //     // expect(actions.storeQuoraData).toHaveBeenCalled();
    //     expect(actions.loginUser).toHaveBeenCalled();
    //     // expect(wrapper.vm.loginHandle).toHaveBeenCalled();
        
    //     expect(wrapper.vm.loginHandle).toHaveBeenCalled();
    // });
});