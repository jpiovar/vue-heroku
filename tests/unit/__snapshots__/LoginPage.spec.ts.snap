// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage.vue should match the snapshot 1`] = `
<div
  class="container"
  id="loginPage"
>
  <div
    class="row align-items-center justify-content-center"
  >
    <div
      class="col-md-5"
    >
      <loginform-stub
        loginref="function () { [native code] }"
      />
       
      <span
        id="footer"
      >
        Juraj Piovar @ 2019
      </span>
    </div>
  </div>
</div>
`;
